src/libraries/System.Security.Cryptography.Encoding/tests/Cbor/CborReader.String.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers.Binary;
+
+namespace System.Security.Cryptography.Encoding.Tests.Cbor
+{
+    internal ref partial struct CborValueReader
+    {
+        private static readonly System.Text.Encoding s_utf8Encoding = new System.Text.UTF8Encoding(encoderShouldEmitUTF8Identifier: false, throwOnInvalidBytes: true);
+
+        // Implements major type 2 decoding per https://tools.ietf.org/html/rfc7049#section-2.1
+        public byte[] ReadByteString()
+        {
+            CborDataItem header = ReadInitialByte(expectedType: CborMajorType.ByteString);
+            int length = checked((int)ReadUnsignedInteger(header, out int additionalBytes));
+            byte[] result = new byte[length];
+            _buffer.Slice(1 + additionalBytes, length).CopyTo(result);
```

Comment:
I don't think we want ArgumentOutOfRangeException here.  Add a test for the length exceeding the remainder of the payload.

Reply:
What's would be the appropriate exception type in this case? FormatException?

Reply:
CryptographicException? Some new CoseException? InvalidOperationException?

It's not 100% important right now, other than a) not an ArgumentException (it's not related to caller arguments) and b) there's a test for it (which can be updated later; probably in bulk if it's not InvalidOperationException).